htdocs/categories/class/categorie.class.php

```diff
@@ -642,17 +642,24 @@ function add_type($obj,$type)
 		}
 		else
 		{
-		    $this->db->rollback();
-			if ($this->db->lasterrno() == 'DB_ERROR_RECORD_ALREADY_EXISTS')
-			{
-				$this->error=$this->db->lasterrno();
-				return -3;
-			}
-			else
-			{
-				$this->error=$this->db->lasterror();
-			}
-			return -1;
+            if ($this->db->lasterrno() == 'DB_ERROR_RECORD_ALREADY_EXISTS')
+            {
+                if (empty($conf->global->CATEGORIE_RECURSIV_ADD)) {
+                    $this->db->rollback();
+                    $this->error = $this->db->lasterrno();
+                    return -3;
+                }
+                else {
+                    $this->db->commit();
```

Comment:
We can't make a commit if a sql error (for exemple because of duplicate key) is thrown. With some postgresql, this will failed.
To solve the bug, we must introduce a check before (if CATEGORIE_RECURSIV_ADD is on) to kown if it already exists.